stdlib/public/core/StringCreate.swift

```diff
@@ -236,6 +262,10 @@ extension String {
     where Input == UnsafeBufferPointer<UInt8>, Encoding == Unicode.ASCII)
   @_specialize(
     where Input == Array<UInt8>, Encoding == Unicode.ASCII)
+  @_specialize(
+    where Input == UnsafeBufferPointer<UInt16>, Encoding == UTF16)
```

Comment:
Not sure about these either

Reply:
As a separate point, we might want a separate fully concrete code path for UTF-16 anyways, rather than going through generic transcoding.